### PR TITLE
Resolution upgrade

### PR DESCRIPTION
**Pull request by thobbsinteractive**

### CODE REVIEW

File: remc2/sub_main.cpp [hunk 1]
```diff
@@ -117827,7 +117817,7 @@ void* sub_83D70_malloc1(int a1)//264d70
 	if ( !v3 || !sub_83E00((int)v3, v4) )
 	  return 0;
 	sub_85350();
-	memset((void*)*v3, 0, v4);//problém
+	memset((void*)*v3, 0, v4);//problÃ©m
```
Comment (GrimSqueaker) [inline remc2/sub_main.cpp:117820]: Is your editor configured to use UTF-8? It looks like there are quite a few changes here that seem to be conversions into an ISO-8859 codepage or so. I think it is just comments but maybe it is worth checking the settings.

File: remc2/engine/engine_support.h [hunk 2]
```diff
@@ -1447,30 +1447,14 @@ typedef struct {//lenght 7
...
-	//Bit8u byte_0x2194;//8596 resolution
-	//Bit8u byte_0x2195;//8597 shadows
 	type_str_0x2196 str_0x2196;
-	//Bit8u byte_0x2196;//8598 lights
-	//Bit8u byte_0x2197;//setting
-	//Bit8u byte_0x2198;//8600 transparency
-	//Bit8u byte_0x2199;//8601 flat
 }
-type_str_0x218A_settings;
+GameSettingsStruct_t;
```
Comment (GrimSqueaker) [inline remc2/engine/engine_support.h:1455]: I like all these renamings that start to make the logic clearer.

File: remc2/sub_main.cpp [hunk 3]
```diff
@@ -117930,47 +117920,12 @@ int sub_84000(int a1)//265000
...
-		x_D41A0_BYTEARRAY_4_struct.byteindex_34 = 1;
-	}
-	else
-	{
-		//result = (int)x_D41A0_BYTEARRAY_4;
-		x_D41A0_BYTEARRAY_4_struct.byteindex_34 = 0;
-	}
-	//return result;
+	//Just assume High spec for now
+	x_D41A0_BYTEARRAY_4_struct.m_wHighSpeedSystem = 5;
```
Comment (GrimSqueaker) [inline remc2/sub_main.cpp:117926]: Makes sense. There is also some code for reading dig.ini and midi.ini as well as dealing with IRQ settings etc. that can just be removed in my opinion.
Moreover, what do you think of removing the 320 graphics mode at all?

**Comment by thobbsinteractive** [comment]
Hi Seb,

I'll check my encoding, normally smart git seems to sort encoding at
check-in, it's likely a GitHub setting. I think it just effects comments.

Weirdly I quite like the 320 setting, but it has little practical use since
I set it to use 640 as a minimum. Even if this ever runs on a phone the
minimum these days will still likely be 640.

In other news, i am a bit torn. I believe I will eventually work out how to
increase the resolution above 640 and increase the draw distance and maybe
even move the render to another thread. But after completing that step I
would ultimately be looking to replace most of the existing engine with
openGL. Not only that, a HDv or 4k software render will be sllloooow on any
machine (unless lots of threading is used (like a GPU essentially does).

So while it makes sense to work out how the engine works and refactor it
away from the game logic, it might NOT make sense to spend too much time
working out how it renders if most of it has to be dumped anyway.

What do you think?

**Comment by GrimSqueaker** [comment]
Hi Tim,

I think that replacing the render engine by OpenGL is a very good idea. But I don't know if I can be of much help in this regard, since my knowledge of graphics programming is really shallow.
Do you already have an idea how to get there?
I guess one prerequisite is to understand and refactor the game logic, data structures, etc. In my previous repo I came to a similar conclusion from a different direction, because this is also a prerequisite to make it build in 64 bit or something else like ARM.
I just thought that every bit of removed code - like the 320 mode - would help to make it easier to understand. But you are right, time is a limited resource and removing the 320 mode is not a good enough reason by itself.

By the way:
I started to look into fixing the sound on Linux and am using the libADLMIDI whose source is checked into the root directory in order to get rid of as many special Linux #ifdef cases as possible. However, even all the commented code on the way to the main menu, I get no sound at all. Neither in the intro movie nor in the main menu.
Does sound work in the Windows version?

One more thing:
Can we use your fork as the de-facto reference repository? What I mean is: I would like to put some ideas, bugs, plans, etc. into "issues" so that we can discuss them and track their progress on a per-topic level. It would make sense to have this in one place only. I am using your fork anyhow all the time to keep up with all changes on the "development" branch. For changes to the code I would keep up my fork, so that I can get messy with branches over there. :-)

Cheers,
Sebastian

**Comment by thobbsinteractive** [comment]
Hi Seb,

Feel free to branch directly from my repo.

Everything works with windows sounds, but there a few small issues (some
ogg music tracks cause performance issues). I would suggest turning of HD
sounds in the config.ini, that could be related to your issue as it uses
"oog" for the music. There is also a fix in that HD code to slow down
converted 22k to 44k sounds, that could also be a cause of some issues.

At some point I will take a proper look at the sound and remove the
conversation. It would be better to fix them externally and make them an
optional extra. The current sound implementation is not great and SDL
should be capable of some much more.

I am making slow progress in translating the graphics code, but I believe I
will figure out the draw distance soon.

Kind regards

Tim

**Comment by GrimSqueaker** [comment]
Hi Tim,

silly me did disable sound in the beginning to get it compiling... Enabling the sound was essentially just an include of "SDL2/SDL_mixer.h".
Now I will continue with fixing the code that does not compile because of illegal type conversions and I'll also compile it with sanitizers enabled to start fixing otherwise problematic code - all with the intention to make it more portable (64 bit, ARM, etc.)

Cheers,
Sebastian

**Pull request merged**